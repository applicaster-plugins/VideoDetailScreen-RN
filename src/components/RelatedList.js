import React, { PureComponent } from 'react';
import { View, Text, ActivityIndicator, ViewPropTypes } from 'react-native';
import PropTypes from 'prop-types';
import RelatedTeaser from './RelatedTeaser';
import { labels } from '../utils';
import { styles } from './RelatedList.styles';

const { bool, array, func } = PropTypes;

class RelatedList extends PureComponent {
  constructor(props) {
    super(props);
    // deferred to load theme parameter
    this.styles = styles();
  }

  render() {
    const { loading, list, onTapItem, style } = this.props;
    return (
      <View style={[this.styles.container, style]}>
        <Text style={this.styles.title}>{labels.relatedVideos}</Text>
        {loading && (
          <View style={this.styles.loader}>
            <ActivityIndicator size="large" />
          </View>
        )}
        {list &&
          list.map(({ id, ...teaser }) => (
            <RelatedTeaser
              key={id}
              style={this.styles.teaser}
              onTap={onTapItem}
              {...teaser}
            />
          ))}
      </View>
    );
  }
}

RelatedList.propTypes = {
  loading: bool,
  list: array,
  onTapItem: func,
  style: ViewPropTypes.style
};

export default RelatedList;
