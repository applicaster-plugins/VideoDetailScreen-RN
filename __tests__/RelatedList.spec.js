import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import RelatedList from '../src/components/RelatedList';

jest.mock('../src/utils/zappConfig');

const style = {
  marginTop: 20,
  marginLeft: 0,
  flex: 0
};

const onTapItem = jest.fn();

const list = [
  {
    id: 'a4d12b904e3a013739347054d2ab776f',
    imageUri:
      'http://manager.vidibuscloud.net/mediafiles/8893/6575/6215/8eb65e3055be0136b7f5543d7edaf0a0/opener-tennis-allgemein_1024x576.jpg',
    title: 'German Masters Series – TVM Open 2019 - Tag 3',
    dateTime: 'HEUTE / 09:00 Uhr',
    duration: false,
    label: undefined,
    sportLabel: 'Tennis',
    data: {
      type: {
        value: 'channel'
      },
      title: 'German Masters Series – TVM Open 2019 - Tag 3',
      id: 'a4d12b904e3a013739347054d2ab776f',
      summary: 'HEUTE / 09:00 Uhr',
      published: '',
      updated: '',
      author: {
        name: ''
      },
      media_group: [
        {
          type: 'image',
          media_item: [
            {
              src:
                'http://manager.vidibuscloud.net/mediafiles/8893/6575/6215/8eb65e3055be0136b7f5543d7edaf0a0/opener-tennis-allgemein_1024x576.jpg',
              key: 'image_base',
              type: 'image'
            }
          ]
        }
      ],
      link: {},
      extensions: {
        section: 'Tennis',
        start_time: '2019-05-09T09:00:00+00:00',
        end_time: '2019-05-09T11:00:00+00:00',
        open_with_plugin_id: 'SDVideoPage',
        is_live: true,
        body_text:
          '*Sportdeutschland.TV* zeigt die German Masters Series - TVM Open 2019 vom 07. bis zum 11.05. aus Troisdorf live und on demand.\n\nHol Dir SPORTDEUTSCHLAND.TV PLUS und genieße unser Angebot komplett werbefrei.'
      },
      content: {
        type: 'video/hls',
        src:
          'http://origin.vidibus.net/live/stream/itf-court-1.m3u8?realm=6e8cba30a0da0130491a6c626d58b44c&asset=966ef6104e3a01372994543d7edaf0a0_2&realm=6e8cba30a0da0130491a6c626d58b44c&asset=966ef6104e3a01372994543d7edaf0a0_2'
      }
    }
  }
];

it('RelatedList renders correctly', () => {
  const tree = renderer
    .create(
      <RelatedList
        loading={false}
        list={list}
        onTapItem={onTapItem}
        style={style}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
