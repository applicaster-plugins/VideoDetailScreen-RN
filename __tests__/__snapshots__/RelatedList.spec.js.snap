// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedList renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "flex": 0,
        "marginLeft": 0,
        "marginTop": 20,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "rgb(255, 255, 255)",
        "fontFamily": "Exo2-SemiBold",
        "fontSize": 20,
        "letterSpacing": -0.4,
        "lineHeight": 26,
        "marginBottom": 20,
      }
    }
  >
    Related Videos
  </Text>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        undefined,
        Object {
          "marginBottom": 24,
        },
      ]
    }
  >
    <View>
      <Image
        source={
          Object {
            "uri": "http://manager.vidibuscloud.net/mediafiles/8893/6575/6215/8eb65e3055be0136b7f5543d7edaf0a0/opener-tennis-allgemein_1024x576.jpg",
          }
        }
      />
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
      >
        HEUTE / 09:00 Uhr
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
      >
        German Masters Series â€“ TVM Open 2019 - Tag 3
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
      >
        Tennis
      </Text>
    </View>
  </View>
</View>
`;
